### DISCUSSION TITLE
module name error

### DISCUSSION THREAD

**Discussion by devops-4u** [explosion/spaCy#7718] [Help: Installation] [answered]
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-1-40326a359aee> in <module>
      2 from chatterbot.trainers import ChatterBotCorpusTrainer
      3 
----> 4 chatbot = ChatBot('bot')
      5 trainer = ChatterBotCorpusTrainer(chatbot)
      6 trainer.train('chatterbot.corgus.english')

C:\ProgramData\Anaconda3\lib\site-packages\chatterbot\chatterbot.py in __init__(self, name, **kwargs)
     26         self.logic_adapters = []
     27 
---> 28         self.storage = utils.initialize_class(storage_adapter, **kwargs)
     29 
     30         primary_search_algorithm = IndexedTextSearch(self, **kwargs)

C:\ProgramData\Anaconda3\lib\site-packages\chatterbot\utils.py in initialize_class(data, *args, **kwargs)
     31         Class = import_module(data)
     32 
---> 33         return Class(*args, **kwargs)
     34 
     35 

C:\ProgramData\Anaconda3\lib\site-packages\chatterbot\storage\sql_storage.py in __init__(self, **kwargs)
     18 
     19     def __init__(self, **kwargs):
---> 20         super().__init__(**kwargs)
     21 
     22         from sqlalchemy import create_engine

C:\ProgramData\Anaconda3\lib\site-packages\chatterbot\storage\storage_adapter.py in __init__(self, *args, **kwargs)
     19 
     20         self.tagger = PosLemmaTagger(language=kwargs.get(
---> 21             'tagger_language', languages.ENG
     22         ))
     23 

C:\ProgramData\Anaconda3\lib\site-packages\chatterbot\tagging.py in __init__(self, language)
     11         self.punctuation_table = str.maketrans(dict.fromkeys(string.punctuation))
     12 
---> 13         self.nlp = spacy.load(self.language.ISO_639_1.lower())
     14 
     15     def get_bigram_pair_string(self, text):

C:\ProgramData\Anaconda3\lib\site-packages\spacy\__init__.py in load(name, **overrides)
     28     if depr_path not in (True, False, None):
     29         warnings.warn(Warnings.W001.format(path=depr_path), DeprecationWarning)
---> 30     return util.load_model(name, **overrides)
     31 
     32 

C:\ProgramData\Anaconda3\lib\site-packages\spacy\util.py in load_model(name, **overrides)
    168             return load_model_from_link(name, **overrides)
    169         if is_package(name):  # installed as package
--> 170             return load_model_from_package(name, **overrides)
    171         if Path(name).exists():  # path to model data directory
    172             return load_model_from_path(Path(name), **overrides)

C:\ProgramData\Anaconda3\lib\site-packages\spacy\util.py in load_model_from_package(name, **overrides)
    188 def load_model_from_package(name, **overrides):
    189     """Load a model from an installed package."""
--> 190     cls = importlib.import_module(name)
    191     return cls.load(**overrides)
    192 

C:\ProgramData\Anaconda3\lib\importlib\__init__.py in import_module(name, package)
    125                 break
    126             level += 1
--> 127     return _bootstrap._gcd_import(name[level:], package, level)
    128 
    129 

C:\ProgramData\Anaconda3\lib\importlib\_bootstrap.py in _gcd_import(name, package, level)

C:\ProgramData\Anaconda3\lib\importlib\_bootstrap.py in _find_and_load(name, import_)

C:\ProgramData\Anaconda3\lib\importlib\_bootstrap.py in _find_and_load_unlocked(name, import_)

ModuleNotFoundError: No module named 'en'

**Comment by devops-4u** [comment]
pls help me as i already install spacy still i got error after that i installed python -m spacy download en i got this-
 Download successful but linking failed
Creating a shortcut link for 'en' didn't work (maybe you don't have admin
permissions?), but you can still load the model via its full package name: nlp =
spacy.load('en_core_web_sm')

**Reply by svlandeg** [reply]
Which version of spaCy are you using? Which models do you have installed? Can you provide the output of running `spacy info` on the command line?

**Reply by svlandeg** [reply]
As a first guess, it looks like you're trying to use a plugin that assumes `spacy.load('en')` works, but this doesn't work on your system because you don't have admin privilidges to create the symlink?

**Comment by svlandeg** [comment]
Hi! As a maintenance note - this type of question is better suited for the discussion forum, so I'll move it there. The original issue will be closed, but will link through to the new, open discussion thread.

**Comment by devops-4u** [comment]
On Fri, 9 Apr 2021 at 3:10 PM, Sofie Van Landeghem ***@***.***>
wrote:

> As a first guess, it looks like you're trying to use a plugin that assumes
> spacy.load('en') works, but this doesn't work on your system because you
> don't have admin privilidges to create the symlink?
>
yes i dnt hv

**Comment by devops-4u** [comment]
now tell me what will i do to remove this error???

**Comment by svlandeg** [comment, marked as answer]
The use of symlinks has been deprecated for some time - instead of writing something like `spacy.load("en")` we advise to be specific about the model you want to use - e.g. `spacy.load("en_core_web_sm")`

From your original post, I get the impression that you're using an external library though. If you can't change the `spacy.load()` statement yourself, you'll have to find a way to ensure the linking works on your system. I assume you're working on Windows? The reference to "admin permissions" means that the system prevented proper execution of the linking. To remedy, you should start your console that you're using to execute `spacy download` with admin priviledges. 

You can find more information and workarounds detailed in the v2 docs here: https://v2.spacy.io/usage/#symlink-privilege
